lib/slim-rails/register_engine.rb

```diff
@@ -0,0 +1,30 @@
+module Slim
+  module Rails
+    module RegisterEngine
+      def self.register_engine(app, config)
+        if ::Rails::VERSION::MAJOR == 3
+          register_engine3(app, config)
+        else
+          register_engine4(app, config)
+        end
+      end
+
+      private
+
+      def self.register_engine3(app, _config)
+        return unless app.assets
+        return unless app.assets.respond_to?(:register_engine)
+
+        app.assets.register_engine('.slim', Slim::Template)
+      end
+
+      def self.register_engine4(_app, config)
```

Comment:
> Yes, register_engine plays the 'dispatcher' role in this scenario.

I understand, thanks. 'underscored' version sounds nice 👍  thank you for suggestion, i'll change the method name and signature then force push squashed recent some commits.